Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -413,7 +413,7 @@ public void RestoreActionStates(UnmanagedMemory oldState)
         /// <param name="actionIndex">Action whose state to reset.</param>
         /// <param name="toPhase">Phase to reset the action to. Must be either <see cref="InputActionPhase.Waiting"/>
         /// or <see cref="InputActionPhase.Disabled"/>. Other phases cannot be transitioned to through resets.</param>
-        public void ResetActionState(int actionIndex, InputActionPhase toPhase = InputActionPhase.Waiting)
+        public void ResetActionState(int actionIndex, InputActionPhase toPhase = InputActionPhase.Waiting, bool hardReset = false)
         {
             Debug.Assert(actionIndex >= 0 && actionIndex < totalActionCount, "Action index out of range when resetting action");
```

Comment:
`Debug.Assert` will just log an assert message but continue so then we hit an exception due to the invalid index. Should we not throw in this case? Maybe use `Assert.IsTrue `so that only 1 error is generated instead of 2

Reply:
>Should we not throw in this case?

IMO, we shouldn't. To me, these things serve two different purposes. Exceptions are aimed at the user of an API whereas Debug.Asserts are aimed at the developer of the API. An exception should trigger from either the user incorrectly using an API whereas Debug.Asserts should only trigger if there's a bug and the code has its own wires crossed.

The xmldocs here are a bit misleading. The APIs here are actually all internal.